### DISCUSSION TITLE
Extended 'Service Timeout' for webhooks

### DISCUSSION THREAD

**Discussion by ishepherd** [community/community#24804] [Apps, API and Webhooks]
<p>I’ve created a Github App that will work on our company’s pull requests.</p>
<p>The Azure Functions serverless platform removes the allocated compute once the app gets cold (zero requests for &gt;15 mins). So there is a cold start delay to get it back. This is only ~10 seconds but that plus Github login time, etc, means the first webhook of the day gets a failed delivery “Service timeout”. (It is still processed fine.)</p>
<ol>
<li>
<p>What is the allowed service timeout?</p>
</li>
<li>
<p>Can we please have a setting to increase it?</p>
</li>
</ol>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hey <a class="mention" href="/ishepherd">@ishepherd</a>,</p>
<p>Thanks for being here!</p>
<p>GitHub expects that integrations respond within 10 seconds of receiving the webhook payload, as seen in the <a href="https://developer.github.com/v3/guides/best-practices-for-integrators/#favor-asynchronous-work-over-synchronous" rel="nofollow noopener">docs.</a></p>
<p>At this time there is no setting to increase it, I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>

**Reply by beliaev-maksim** [reply]
here is the new link for docs
https://docs.github.com/en/webhooks/using-webhooks/best-practices-for-using-webhooks#respond-within-10-seconds

**Comment by ishepherd** [comment]
<p>Thanks <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> !<br>
Yes, please consider it. FAAS platforms are perfect for cheap/free GitHub webhook implementations - apart from this one small issue.</p>

**Comment by samuelcolvin** [comment]
<p>To work around this I’ve build a proxy using cloudflare workers, see <a href="https://github.com/samuelcolvin/cloudflare-proxy" title="samuelcolvin/cloudflare-proxy" rel="nofollow noopener">samuelcolvin/cloudflare-proxy</a>. The point is it responds immediately (or as soon as it’s ready the request body), but continues to make the request in the background, it works fine with requests taking ~30seconds, probably significantly longer.</p>
<p>You can use it by making a request to</p>
<pre><code>https://proxy.samuelcolvin.workers.dev?upstream=&lt;webhook endpoint&gt;
</code></pre>
<p>e.g.</p>
<pre><code>curl -v https://proxy.samuelcolvin.workers.dev -G --data-urlencode 'upstream=https://exampl.com/endpoint'
</code></pre>
<p>You can either use “<a href="https://proxy.samuelcolvin.workers.dev" rel="nofollow noopener">https://proxy.samuelcolvin.workers.dev</a>” (I have no intention of changing it or adding logging) or deploy the worker yourself using the code linked to above.</p>
<p>(by the way AndreaGriffiths11, you can’t really mark this as “solved”, when what you’ve actually said is “won’t fix”).</p>
